token: fix floating point accuracy in C backend generation

chqrlie opened the pull request:
* use `ftoa()` to ensure reversible floating point number conversions

bvdberg commented:
merged. I changed 2 things, removed 'default' case and by adding 'case Default'. Also remove unnecessary {} in case statement. In C2 you can do:
case 1:
   i32 a = 10;

chqrlie commented:
> merged. I changed 2 things, removed 'default' case and by adding 'case Default'. Also remove unnecessary {} in case statement. In C2 you can do: ```case 1: i32 a = 10;```

I got compiler errors for this or ```label: i32 a = 10;```.  Do you rely on a gcc extension to support labelled definitions?

bvdberg commented:
No, c2c will check if the first statement in a switch is a declaration and then wrap the body with { .. }
case 1: {
  <body>
}

bvdberg commented:
c2c does not generally add braces, only for case statements..

chqrlie commented:
> c2c does not generally add braces, only for case statements..

Of course, `c2c` should only add braces as needed.  Labelled definitions need them for the same reason case statements need them.
